### ISSUE TITLE
Not passing 'liveEndpointUrlPrefix' to client in live environment results in ambiguous error

### ISSUE THREAD

**Issue by GonathanJawecki** [Adyen/adyen-dotnet-api-library#109]
Hey, just recently encountered this small issue when I was using your library.

When newing up a Client class I was using the constructor: 
```c# 
Client(string xapikey, Environment environment) 
```
for both Test and Live environments, this worked fine for Test.

However, for Live I was getting this error when I was calling the `Checkout.PaymentMethods` method:

```c#
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString)
   at System.Net.WebRequest.Create(String requestUriString)
   at Adyen.EcommLibrary.HttpClient.HttpUrlConnectionClient.GetHttpWebRequest(String endpoint, Config config, Boolean isApiKeyRequired, RequestOptions requestOptions)
   at Adyen.EcommLibrary.HttpClient.HttpUrlConnectionClient.Request(String endpoint, String json, Config config, Boolean isApiKeyRequired, RequestOptions requestOptions)
   at Adyen.EcommLibrary.Service.ServiceResource.Request(String json, RequestOptions requestOptions)
   at Adyen.EcommLibrary.Service.Checkout.PaymentMethods(PaymentMethodsRequest paymentMethodsRequest)
```
Obviously this was because I wasn't using the constructor:
```c# 
Client(string xapikey, Environment environment, string liveEndpointUrlPrefix) 
``` 
to declare the data center correctly for Live. However, this wasn't picked up until we released to our Staging environment. As the Live URL was malformed since it was missing the `liveEndpointUrlPrefix`.

I feel that ideally it would be nice to have had a more descriptive error before the code made it to a method call. I don't really understand the reason behind the config value not setting the `CheckoutEndpoint` if the `liveEndpointUrlPrefix` is null in this switch statement? is this Possibly a bug?

```c#
switch (environment)
{
    case Environment.Test:
        Config.Endpoint = ClientConfig.EndpointTest;
        Config.HppEndpoint = ClientConfig.HppTest;
        Config.CloudApiEndPoint = ClientConfig.CloudApiEndPointTest;
        Config.CheckoutEndpoint = ClientConfig.CheckoutEndpointTest;
        break;
    case Environment.Live:
        Config.Endpoint = ClientConfig.EndpointLive;
        Config.HppEndpoint = ClientConfig.HppLive;
        Config.CloudApiEndPoint = ClientConfig.CloudApiEndPointLive;
        //set live endpoint for checkout api
        if (!string.IsNullOrEmpty(liveEndpointUrlPrefix))
        {
            Config.Endpoint = ClientConfig.EndpointProtocol + liveEndpointUrlPrefix + ClientConfig.EndpointLiveSuffix;
            Config.CheckoutEndpoint = ClientConfig.EndpointProtocol + liveEndpointUrlPrefix + ClientConfig.CheckoutEndpointLiveSuffix;
        }
        break;
}
```

If this is intended, please dirsregard. I'm happy to push a fix if you think it should throw here, or use a default `CheckoutEndpoint` instead? Thanks for taking the time to humor my issue and thanks for a great library!

**Comment by AlexandrosMor** [comment]
Hello @GonathanJawecki ,

Thank you for the description. I agree the error message can be more clear. It would be great if you can push the fix  :) . Please add also a small test for it.

Thank you in advance,
Alexandros Moraitis

**Comment by GonathanJawecki** [comment]
Thanks @AlexandrosMor,

No worries, I've knocked up a pull request here: https://github.com/Adyen/adyen-dotnet-api-library/pull/112 and added unit tests around my code change.

let me know if that's acceptable.

Cheers,
Jonathan

**Comment by AlexandrosMor** [comment]
fixed #112

**Comment by jwsadler** [comment]
Which version is this fix in????
